pkg/cmd/repo/archive/archive_test.go

```diff
@@ -66,80 +51,85 @@ func TestNewCmdArchive(t *testing.T) {
 				return
 			}
 			assert.NoError(t, err)
-			assert.Equal(t, tt.output.RepoArg, gotOpts.RepoArg)
 		})
 	}
 }
 
 func Test_ArchiveRun(t *testing.T) {
+	queryResponse := `{ "data": { "repository": 
+							{ "id": "THE-ID",
+					   		"isArchived": %s} 
+						} }`
 	tests := []struct {
 		name       string
 		opts       ArchiveOptions
 		httpStubs  func(*httpmock.Registry)
+		askStubs   func(*prompt.AskStubber)
 		isTTY      bool
 		wantStdout string
 		wantStderr string
 	}{
 		{
 			name:       "unarchived repo tty",
-			opts:       ArchiveOptions{RepoArg: "OWNER/REPO"},
 			wantStdout: "✓ Archived repository OWNER/REPO\n",
-			isTTY:      true,
+			askStubs: func(q *prompt.AskStubber) {
+				q.StubOne(true)
+			},
+			isTTY: true,
+			opts:  ArchiveOptions{RepoArg: "OWNER/REPO"},
 			httpStubs: func(reg *httpmock.Registry) {
 				reg.Register(
 					httpmock.GraphQL(`query RepositoryInfo\b`),
-					httpmock.StringResponse(`{ "data": { "repository": {
-												"id": "THE-ID",
-												"isArchived": false} } }`))
+					httpmock.StringResponse(fmt.Sprintf(queryResponse, "false")))
 				reg.Register(
 					httpmock.GraphQL(`mutation ArchiveRepository\b`),
 					httpmock.StringResponse(`{}`))
 			},
 		},
 		{
-			name:  "unarchived repo notty",
-			opts:  ArchiveOptions{RepoArg: "OWNER/REPO"},
-			isTTY: false,
+			name:       "infer base repo",
+			wantStdout: "✓ Archived repository OWNER/REPO\n",
+			opts:       ArchiveOptions{},
+			askStubs: func(q *prompt.AskStubber) {
+				q.StubOne(true)
+			},
+			isTTY: true,
 			httpStubs: func(reg *httpmock.Registry) {
 				reg.Register(
 					httpmock.GraphQL(`query RepositoryInfo\b`),
-					httpmock.StringResponse(`{ "data": { "repository": {
-												"id": "THE-ID",
-												"isArchived": false} } }`))
+					httpmock.StringResponse(fmt.Sprintf(queryResponse, "false")))
 				reg.Register(
 					httpmock.GraphQL(`mutation ArchiveRepository\b`),
 					httpmock.StringResponse(`{}`))
 			},
 		},
 		{
 			name:       "archived repo tty",
-			opts:       ArchiveOptions{RepoArg: "OWNER/REPO"},
 			wantStderr: "! Repository OWNER/REPO is already archived\n",
-			isTTY:      true,
-			httpStubs: func(reg *httpmock.Registry) {
-				reg.Register(
-					httpmock.GraphQL(`query RepositoryInfo\b`),
-					httpmock.StringResponse(`{ "data": { "repository": {
-												"id": "THE-ID",
-												"isArchived": true } } }`))
-			},
-		},
-		{
-			name:       "archived repo notty",
 			opts:       ArchiveOptions{RepoArg: "OWNER/REPO"},
-			isTTY:      false,
-			wantStderr: "! Repository OWNER/REPO is already archived\n",
+			askStubs: func(q *prompt.AskStubber) {
+				q.StubOne(true)
+			},
```

Comment:
If the repo is archived we don't actually prompt the user, so this stub is unnecessary.